File: Stronghold.Core/Entities/MembershipPackage.cs
namespace Stronghold.Core.Entities;

public class MembershipPackage : BaseEntity
{
    public string Name { get; set; } = null!;
    public decimal Price { get; set; }
    public int DurationDays { get; set; }
    public string Description { get; set; } = null!;

    public ICollection<Membership> Memberships { get; set; } = new List<Membership>();
}

File: UI/stronghold_desktop/lib/models/membership_package.dart
class MembershipPackage {
  final int id;
  final String name;
  final double price;
  final int durationDays;
  final String description;

  MembershipPackage({
    required this.id,
    required this.name,
    required this.price,
    required this.durationDays,
    required this.description,
  });

  factory MembershipPackage.fromJson(Map<String, dynamic> json) => MembershipPackage(
        id: json['id'] as int,
        name: json['name'] as String,
        price: (json['price'] as num).toDouble(),
        durationDays: json['durationDays'] as int,
        description: json['description'] as String,
      );
}
